flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -354,4 +404,64 @@ public void operationComplete(ChannelFuture future) throws Exception {
 			}
 		}
 	}
+
+	private static class RegisterPeriodicFlushEvent {
+		private final NetworkSequenceViewReader reader;
+		private final long flushTimeout;
+
+		public RegisterPeriodicFlushEvent(NetworkSequenceViewReader reader, long flushTimeout) {
+			this.reader = checkNotNull(reader);
+			this.flushTimeout = flushTimeout;
+		}
+
+		public NetworkSequenceViewReader getReader() {
+			return reader;
+		}
+
+		public long getFlushTimeout() {
+			return flushTimeout;
+		}
+	}
+
+	private static class PeriodicFlushes {
+		private final Map<Long, List<NetworkSequenceViewReader>> periodicFlushes = new HashMap<>();
+		private final Map<NetworkSequenceViewReader, Long> flushTimeouts = new HashMap<>();
+		private final Map<Long, ScheduledFuture<?>> scheduledEvents = new HashMap<>();
+
+		public void register(ChannelHandlerContext ctx, long flushTimeout, NetworkSequenceViewReader reader) {
+			checkState(!flushTimeouts.containsKey(reader));
+			checkState(flushTimeout > 0);
```

Comment:
If we check the `flushTimeout` at the beginning in `ResultPartition` stack as mentioned above, we need not care about it in the following processes.